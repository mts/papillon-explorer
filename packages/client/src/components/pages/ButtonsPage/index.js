import React from 'react';
import { Link } from 'react-router-dom';

const ButtonsPage = () => {
  return (
    <div>
      <h2>Buttons Page</h2>
      <div>
        Under construction ðŸ™‚
        <Link to="/">Back to Home Page</Link>
      </div>
    </div>
  )
}

export default ButtonsPage;
