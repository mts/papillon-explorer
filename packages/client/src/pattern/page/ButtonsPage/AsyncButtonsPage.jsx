import React from 'react'

const LazyButtonsPage = React.lazy(() => import('./ButtonsPage'))

export const AsyncButtonsPage = () => (
  <React.Suspense fallback={null}>
    <LazyButtonsPage />
  </React.Suspense>
)
