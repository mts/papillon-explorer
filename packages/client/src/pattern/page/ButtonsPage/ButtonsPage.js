import React from 'react'
import { Query } from 'react-apollo'
import { startPageRoute } from '../../../routes'
import { PAGES } from '../../../graphql/query'
import ButtonsTemplate from '../../template/ButtonsTemplate'

const ButtonsPage = () => {
  return (
    <Query query={PAGES}>
      {({ data }) =>
        data.pages && data.pages.pages ? (
          <ButtonsTemplate page={data.pages.pages.find(page => page.id === 'buttons')} startPageRoute={startPageRoute} />
        ) : null
      }
    </Query>
  )
}

export default ButtonsPage
