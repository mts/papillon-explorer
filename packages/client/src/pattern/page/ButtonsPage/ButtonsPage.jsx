import React from 'react'
import { useQuery } from '@apollo/react-hooks'
import { PAGES } from '../../../graphql/query'
import { ButtonsTemplate } from '../../template/ButtonsTemplate'

export const ButtonsPage = () => {
  const { data } = useQuery(PAGES)

  if (data && data.pages && data.pages.pages) {
    return <ButtonsTemplate page={data.pages.pages.find(page => page.id === 'buttons')} />
  }

  return null
}

export default ButtonsPage
