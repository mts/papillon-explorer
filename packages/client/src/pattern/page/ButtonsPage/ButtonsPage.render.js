import React from 'react'
import { ButtonsPage } from './ButtonsPage'

export const buttonsPageDefault = <ButtonsPage />
