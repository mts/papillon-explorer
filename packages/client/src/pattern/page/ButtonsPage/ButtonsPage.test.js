import React from 'react'
import { buttonsPageDefault } from './ButtonsPage.render'

jest.mock('@apollo/react-hooks', () => {
  return {
    useQuery: () => {
      return {
        data: {
          pages: {
            pages: [{ id: 'buttons', name: 'some-name' }],
          },
        },
      }
    },
  }
})

jest.mock('react-router-dom', () => ({ Link: () => <div>some link</div> }))

describe('<ButtonsPage />', () => {
  describe('Snaphot', () => {
    test('must match buttonsPageDefault', () => {
      expect(global.renderToJSON(buttonsPageDefault)).toMatchSnapshot()
    })
  })
})
