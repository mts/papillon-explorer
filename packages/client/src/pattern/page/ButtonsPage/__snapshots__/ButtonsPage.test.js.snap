// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonsPage /> Snaphot must match buttonsPageDefault 1`] = `
<div>
  <h2>
    some-name
  </h2>
  <div>
    Under construction
     
    <span
      aria-label="emoji"
      role="img"
    >
      ðŸ™‚
    </span>
    <div>
      some link
    </div>
  </div>
</div>
`;
